code/Weapons/Weapon.cs

```diff
@@ -99,11 +98,12 @@ public void OnDeploy( Grub grub )
 		}
 
 		if ( FiringType is FiringType.Cursor )
-			SetPointerEvents( To.Single( Grub ), true );
+			Event.Run( GrubsEvent.Player.PointerEventChanged, true );
 	}
 
-	public void Holster()
+	public void Holster( Grub _ )
 	{
+		SetParent( null );

```

Comment:
Not sure if this is needed / we should include this

Reply:
I think it's redundant to include. When we parent on deploy it's just to set it up for the first time - it should stay parented no matter what.